### PR TITLE
warn of booleans in constraintset

### LINKED ISSUE #816: Accidental use of posynomial equality should raise a more useful error?
It might be useful if the following raised a more precise error than it currently does, in case a user accidentally makes a posynomial equality constraint. The current error is not particularly helpful.

```python
In [1]: from gpkit import Variable, Model

In [2]: x = Variable('x')

In [3]: y = Variable('y')

In [4]: z = Variable('z')

In [5]: m = Model(z, [z >= x + y, x >= 5, y >= 6]) # what it's meant to be

In [6]: m = Model(z, [z == x + y, x >= 5, y >= 6]) # careless mistake

In [7]: m.solve()
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-1223da3220cb> in <module>()
----> 1 m.solve()

C:\Users\philippe.kirschen\AppData\Local\Continuum\Anaconda2\lib\site-packages\gpkit\constraints\prog_factories.pyc in solvefn(self, solv
er, verbosity, skipsweepfailures, *args, **kwargs)
     77                       solver, verbosity-1, *args, **kwargs)
     78         else:
---> 79             self.program, solvefn = genfunction(self, verbosity-1)
     80             result = solvefn(solver, verbosity-1, *args, **kwargs)
     81             solution.append(result)

C:\Users\philippe.kirschen\AppData\Local\Continuum\Anaconda2\lib\site-packages\gpkit\constraints\prog_factories.pyc in programify(self, v
erbosity, substitutions, **kwargs)
     32         if not substitutions:
     33             substitutions = self.substitutions
---> 34         prog = program(self.cost, self, substitutions, verbosity, **kwargs)
     35         if return_attr:
     36             return prog, getattr(prog, return_attr)

C:\Users\philippe.kirschen\AppData\Local\Continuum\Anaconda2\lib\site-packages\gpkit\geometric_program.pyc in __init__(self, cost, constr
aints, substitutions, verbosity)
     64
     65         for constraint in constraints:
---> 66             constraint.substitutions = self.substitutions
     67             try:
     68                 constr_posys = constraint.as_posyslt1()

AttributeError: 'bool' object has no attribute 'substitutions'

```

### PR DESCRIPTION
**Pull request by bqpd**
closes #816

### CODE REVIEW

File: gpkit/constraints/set.py [hunk 1]
```diff
@@ -30,6 +30,16 @@ def __init__(self, constraints, substitutions=None, recursesubs=True):
                 if (hasattr(constraint, "__iter__") and
                         not isinstance(constraint, ConstraintSet)):
                     self[i] = ConstraintSet(constraint)
+                elif isinstance(constraint, bool):
+                    if i == 0:
+                        adjacent = "at the start, before %s" % self[i+1]
+                    elif i == len(self) - 1:
+                        adjacent = "at the end, after %s" % self[i-1]
```
Comment (whoburg) [inline gpkit/constraints/set.py]: hmm, won't these cause an IndexError if there is only a single constraint?

**Comment by whoburg** [comment]
review complete.

BTW, this is making me wonder whether we should implement `Nomial.__eq__` to return a Constraint (consistent with other operators we've overloaded), and then call for `__nonzero__` when we explicitly want it, e.g. in unit tests. This is clearly a separate design discussion though.

**Comment by bqpd** [comment]
Good question. That might lead users to expect PosynomialEqualitites to be overloaded as well?

I'll open that issue. Is this ready to merge?

File: gpkit/constraints/set.py (hunk 1 above)
Comment (bqpd) [inline gpkit/constraints/set.py]: Good call.

**Comment by bqpd** [comment]
Addressed review.

**Comment by whoburg** [comment]
@bqpd I'm ready to merge this -- I'll let you do it since I added a commit. Consider rebasing to squash the three commits into one.

**Pull request merged**